### PR TITLE
Add --domain flag to project init

### LINKED ISSUE #334: Optional domain name while initializing a project
## Story
As a user that wants to create web applications under my domain name,   
I want the `"project"` command to take in my domain name as an option,  
So that web applications under my  `{env}` environment are available at `"https://{project}-{env}.{domainName}"`

### User Experience
An **optional** flag for `"project init"`:
```sh
--domain string          Optional. Your existing custom domain name.
```
By default the domain name will be empty for the project.

In the future, we can provide users a separate command `archer domain add -n amazon.com -p smile` that has an interactive experience to add a domain name to an existing project. 

#### Errors
If there is a `ErrCodeUnsupportedTLD` from [route53domains.GetDomainDetail](https://docs.aws.amazon.com/sdk-for-go/api/service/route53domains/#Route53Domains.GetDomainDetail), the error message will be:
```
domain efekarakus.com is not registered with route53 in account 11111111111
```
Otherwise, we'll display the error as:
```
get domain details for efekarakus.com: {someApiError}
```

### Programming Model
This will add an additional field to the `archer.Project` struct named `Domain`

### PR DESCRIPTION
**Pull request by efekarakus**
Resolves #334 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/store/ssm/ssm.go [hunk 1]
```diff
@@ -66,6 +69,9 @@ func NewStore() (*SSM, error) {
 	return &SSM{
 		systemManager: ssm.New(sess),
 		identity:      identity.New(sess),
+		// If the region is not set to us-east-1, we can't find out if a domain name exists in the account.
```
Comment (SoManyHs) [inline internal/pkg/store/ssm/ssm.go]: wat =P

File: internal/pkg/store/ssm/ssm.go [hunk 2]
```diff
@@ -66,6 +69,9 @@ func NewStore() (*SSM, error) {
...
+		// See https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/DNSLimitations.html#limits-service-quotas
+		domains:       route53domains.New(sess, aws.NewConfig().WithRegion("us-east-1")),
```
Comment (SoManyHs) [inline internal/pkg/store/ssm/ssm.go]: It seems a little weird that there is what seems like a client for another service within the SSM client here, but not sure if there's another way...

Comment (efekarakus) [reply internal/pkg/store/ssm/ssm.go]: yeah I agree maybe @kohidave can chime in, but I think the original goal behind the `store` package is for it to be agnostic of which AWS service we use. 

Maybe we can refactor by getting rid of the `ssm` sub-package and moving these files one level above (in a follow-up PR).

What do you think ?

File: internal/pkg/store/ssm/ssm.go (hunk 1 above)
Comment (sonofachamp) [reply internal/pkg/store/ssm/ssm.go]: I think what we're trying to say here is Route 53 Domains is only available in `us-east-1`?

File: internal/pkg/store/ssm/ssm.go (hunk 2 above)
Comment (sonofachamp) [reply internal/pkg/store/ssm/ssm.go]: I think `ssm` and `route53domains` should be isolated and wrapped similarly to some of our other AWS dependencies and the `store` package should orchestrate dependencies to do whatever it needs.

Comment (efekarakus) [reply internal/pkg/store/ssm/ssm.go]: I added an additional commit to remove the "ssm" package by moving this functionality into one level above (the "store" package)

**Pull request merged**